### PR TITLE
development: add development environment docs

### PR DESCRIPTION
**Pull request by bkayranci**
Fluent-bit will provide isolated environment for development. Devcontainer is added as an alternative to Vagrant. The document about creating development environment is added and refered development guide repos.

- [Devcontainer Support PR](https://github.com/fluent/fluent-bit/pull/6880)

### CODE REVIEW

File: development/developer-guide.md [hunk 1]
```diff
@@ -312,6 +312,12 @@ Output plugins are defined in [flb\_output.h](https://github.com/fluent/fluent-b
 
 The [stdout plugin](https://github.com/fluent/fluent-bit/tree/master/plugins/out_stdout) is very simple; review its code to understand how output plugins work.
 
+## Development Environment
+
+Fluent-bit provides an isolated environment from local environment for development. Thanks to that, developer who use different OSes / distros do not have to try finding available required libraries / dependencies for local environment. Development environments of fluent-bit provides required library / tools for development.
```
Comment (patrick-stephens) [inline development/developer-guide.md:317]: Please use separate lines for each sentence - it makes commenting easier.

```suggestion
Fluent-bit provides a standalone environment for development. 
Developers who use different OS or distributions can develop on a simple, common stack. 
The development environment provides the required libraries and tools for you.
```

File: development/developer-guide.md [hunk 2]
```diff
@@ -312,6 +312,12 @@ Output plugins are defined in [flb\_output.h](https://github.com/fluent/fluent-b
...
+
+Development environments by fluent-bit provided are able to create via [Devcontainer](https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#devcontainer) or [Vagrant](https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#vagrant).
```
Comment (patrick-stephens) [inline development/developer-guide.md:319]: Inconsistent `fluent-bit` vs `Fluent-bit`

Comment (patrick-stephens) [reply development/developer-guide.md:319]:
```suggestion
Development environments provided for
- [Devcontainer](https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#devcontainer)
- [Vagrant](https://github.com/fluent/fluent-bit/blob/master/DEVELOPER_GUIDE.md#vagrant).
```

**Review by patrick-stephens — CHANGES_REQUESTED** [review]

**Comment by bkayranci** [comment]
@patrick-stephens thank you for your review :rocket:

**Review by patrick-stephens — APPROVED** [review]

**Comment by patrick-stephens** [comment]
@bkayranci I cannot merge as some of the commits are not signed...

**Comment by bkayranci** [comment]
> @bkayranci I cannot merge as some of the commits are not signed...

Which commits were not signed? [The DCO test](https://github.com/fluent/fluent-bit-docs/pull/1055/checks) is passing. There are "Signed-off-by" on  all commits.

**Comment by bkayranci** [comment]
@patrick-stephens I found and fixed the not signed commit :+1: 

Could you try to merge again please? :rocket:

**Pull request merged**